packages/vc-verification/src/role-resolvers.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  IIssuerDefinition,
+  IRoleDefinitionV2,
+} from '@energyweb/credential-governance';
+import { providers } from 'ethers';
+import { RoleDefinitionResolverV2__factory } from '@energyweb/credential-governance/ethers/factories/RoleDefinitionResolverV2__factory';
+import { RoleDefinitionResolverV2 } from '@energyweb/credential-governance/ethers/RoleDefinitionResolverV2';
+
+abstract class RoleResolver {
+  abstract getRoleIssuers(namespace: string): Promise<IIssuerDefinition>;
+}
+
+export class CachedRoleDefinitionResolver extends RoleResolver {
+  private _roleDefinitions: IRoleDefinitionV2[];
+
+  constructor(roleDefinitions: IRoleDefinitionV2[]) {
+    super();
+    this._roleDefinitions = roleDefinitions;
+  }
+
+  /**
+   *
+   * @param namespace
+   * @returns issuers for the provided namespace form the cached RoleDefinitions
+   */
+  async getRoleIssuers(namespace: string): Promise<IIssuerDefinition> {
+    let issuers: IIssuerDefinition = {};
+    for (let role of this._roleDefinitions) {
+      if (role.roleName == namespace) {
+        issuers = role.issuer;
+      }
+    }
+    return issuers;
+  }
+}
```

Comment:
I don't think it's necessary for ew-credentials to provide an implementation of `CachedRoleDefinitionResolver` as PDA/ICL/ssi-hub, will provide their own implementations as necessary.